### PR TITLE
[ENG-5988] Zotero Implementation for CitationInterface

### PR DESCRIPTION
**Pull request by ly-mariia**
- implemented **Zotero API** and all necessary citations support;

ticket: https://openscience.atlassian.net/jira/software/c/projects/ENG/boards/145?assignee=712020%3A8af272bd-2e97-46c2-b42c-1c046bb6bb57&selectedIssue=ENG-5988

### CODE REVIEW

File: addon_service/migrations/0001_initial.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Generated by Django 4.2.7 on 2024-08-09 15:34
+# Generated by Django 4.2.7 on 2024-09-09 11:49
```
Comment (brianjgeiger) [inline addon_service/migrations/0001_initial.py:1]: Can we not merge into the initial migration anymore? Now that we are on staging, we should be just making normal forward migrations from now on.

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
Overall, I think it looks good, from the perspective of someone who's never implemented anything in GV before. There's a couple of nits, a question that may lead to some work, and a request to not mess with the initial migration anymore.

File: addon_imps/citations/zotero_org.py [hunk 2]
```diff
@@ -1,6 +1,62 @@
...
+        ) as response:
+            if response.status == 200:
+                key_info = await response.json_content()
+                user_id = key_info.get("userID")
+                if not user_id:
+                    raise KeyError("Failed to fetch user ID from Zotero.")
+                return str(user_id)
+            else:
+                raise ValueError(
+                    f"Failed to fetch key information. Status code: {response.status}"
```
[the remark this answers is not in the document]
Comment (ly-mariia) [reply addon_imps/citations/zotero_org.py:26]: Yes, the `Zotero API `does send useful `messages` along with their `status codes`, especially when an `error` occurs. These messages are included in the body of the response and provide additional details about the issue encountered.
I will add them!

**Review by brianjgeiger — APPROVED** [review]
Looks good, thanks!

**Comment by brianjgeiger** [comment]
Well, except for the PR migration conflict.

**Comment by ly-mariia** [comment]
TY, now trying to fix the conflict of migrations

**Pull request merged**